pkg/internal/parser.go

```diff
@@ -162,37 +152,280 @@ func sanitize(groupName string) string {
+	Method      *string
+	Verb        *string
+	Subresource *string
+	Input       *string
+	Result      *string
+}
+
+func NewAPI(
+	root *loader.Package,
+	info *markers.TypeInfo,
+	group, version, applyconfigurationpkg string,
+	namespaceScoped bool,
+	additionalMethods []AdditionalMethod,
+	importList *[]string,
+	skipVerbs, onlyVerbs []string,
+	noVerbs, hasStatus bool,
+	w io.Writer,
+) (*api, error) {
 	typeInfo := root.TypesInfo.TypeOf(info.RawSpec.Name)
 	if typeInfo == types.Typ[types.Invalid] {
 		return nil, fmt.Errorf("unknown type: %s", info.Name)
 	}
 
 	api := &api{
-		Name:         info.RawSpec.Name.Name,
-		Version:      version,
-		PkgName:      group,
-		writer:       w,
-		IsNamespaced: isNamespaced,
-		HasStatus:    hasStatus,
+		Name:                      info.RawSpec.Name.Name,
+		PkgName:                   group,
+		Version:                   version,
+		ApplyConfigurationPackage: applyconfigurationpkg,
+		IsNamespaced:              namespaceScoped,
+		AdditionalMethods:         additionalMethods,
+		importList:                importList,
+		SkipVerbs:                 skipVerbs,
+		NoVerbs:                   noVerbs,
+		HasStatus:                 hasStatus,
+		writer:                    w,
 	}
 
 	api.setCased()
 	return api, nil
 }
 
+// TODO: add templating logic for additional methods
 func (a *api) WriteContent() error {
-	templ, err := template.New("wrapper").Parse(wrapperMethodsTempl)
+	generateApply := len(a.ApplyConfigurationPackage) > 0
+	// Add common template followed by specific methods which are required.
+	if err := templateExecute(wrapperMethodsTempl, *a); err != nil {
+		return err
+	}
+
+	if a.NoVerbs {
+		return nil
+	}
+
+	if a.hasVerb("get") {
+		if err := templateExecute(getTemplate, *a); err != nil {
+			return err
+		}
+	}
+
+	if a.hasVerb("list") {
+		if err := templateExecute(listTemplate, *a); err != nil {
+			return err
+		}
+	}
+
+	if a.hasVerb("watch") {
+		if err := templateExecute(watchTemplate, *a); err != nil {
+			return err
+		}
+	}
+
+	if a.hasVerb("create") {
+		if err := templateExecute(createTemplate, *a); err != nil {
+			return err
+		}
+	}
+
+	if a.hasVerb("update") {
+		if err := templateExecute(updateTemplate, *a); err != nil {
+			return err
+		}
+	}
+
+	if a.hasVerb("updateStatus") && a.HasStatus {
+		if err := templateExecute(updateStatusTemplate, *a); err != nil {
+			return err
+		}
+	}
+
+	if a.hasVerb("delete") {
+		if err := templateExecute(deleteTemplate, *a); err != nil {
+			return err
+		}
+	}
+
+	if a.hasVerb("deleteCollection") {
+		if err := templateExecute(deleteCollectionTemplate, *a); err != nil {
+			return err
+		}
+	}
+
+	if a.hasVerb("patch") {
+		if err := templateExecute(patchTemplate, *a); err != nil {
+			return err
+		}
+	}
+
+	if a.hasVerb("apply") && generateApply {
+		*a.importList = append(*a.importList, util.ImportFormat(fmt.Sprintf("%sapply%s", a.PkgName, a.Version), fmt.Sprintf("%s/%s/%s", a.ApplyConfigurationPackage, a.PkgName, a.Version)))
```

Comment:
It doesn't look like we need this to be a slice pointer.

Reply:
If you want to mutate it in these callers, you need the pointer.